src/main/java/pages/UserTable.java

```diff
@@ -0,0 +1,91 @@
+package pages;
+
+import io.cucumber.datatable.DataTable;
+import java.util.List;
+import java.util.Map;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.Assert;
+import pages.components.HeaderMenu;
+import pages.components.LeftSideMenu;
+import pages.components.RightSideBar;
+
+public class UserTable {
+
+    private HeaderMenu headerMenu;
+    private LeftSideMenu leftSideMenu;
+    private RightSideBar rightSideBar;
+
+    @FindBy(css = "table#user-table select")
+    private List<WebElement> tableDropdowns;
+
+    @FindBy(css = "table#user-table a")
+    private List<WebElement> tableUsernames;
+
+    @FindBy(css = "table#user-table input")
+    private List<WebElement> tableCheckbox;
+
+    @FindBy(xpath = "//td[1]")
+    private List<WebElement> tableNumbers;
+
+    @FindBy(css = "td a")
+    private List<WebElement> tableNames;
+
+    @FindBy(css = "td span")
+    private List<WebElement> tableDescriptions;
+
+    @FindBy(css = "tr > td > select")
+    private List<WebElement> tableDropdown;
+
+    @FindBy(id = "ivan")
+    private WebElement checkboxForIvan;
+
+    public UserTable(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        headerMenu = new HeaderMenu(driver);
+        leftSideMenu = new LeftSideMenu(driver);
+        rightSideBar = new RightSideBar(driver);
+    }
+
+    public void checkTableItems(int numberOfItems, String itemsForCheck) {
+        if (itemsForCheck.toLowerCase().contains("dropdowns")) {
+            Assert.assertEquals(numberOfItems, tableDropdowns.size());
+        } else if (itemsForCheck.toLowerCase().contains("usernames")) {
+            Assert.assertEquals(numberOfItems, tableUsernames.size());
+        } else if (itemsForCheck.toLowerCase().contains("description")) {
+            Assert.assertEquals(numberOfItems, tableDescriptions.size());
+        } else if (itemsForCheck.toLowerCase().contains("checkboxes")) {
+            Assert.assertEquals(numberOfItems, tableCheckbox.size());
+        }
+    }
+
+    public void verifyUserInfoTable(DataTable data) {
+        List<Map<String, String>> table = data.asMaps(String.class, String.class);
+        for (int i = 0; i < table.size(); i++) {
+            Assert.assertEquals(tableNumbers.get(i).getText(), table.get(i).get("Number").trim());
+            Assert.assertEquals(tableNames.get(i).getText(), table.get(i).get("User").trim());
+            Assert.assertEquals(tableDescriptions.get(i).getText().replaceAll("\\s", " "),
+                    table.get(i).get("Description").trim());
+        }
```

Comment:
SoftAssert can be used here so we could check all test data during one test

Reply:
SoftAsserts are used

Reply:
1. You initialize `softAssert` as the class field but it is better to initialize it in the beginning of the assertion method
2. `softAssert.assertAll()` method should be placed outside of the `for` loop because in this implementation it will fail the test after first failed iteration

Reply:
changed, thanks :)